### PR TITLE
Search suggestion

### PR DESCRIPTION
**Pull request by jasontsh**
NOT DONE YET.  #320  #305 #225

### CODE REVIEW

**Comment by dlakata** [comment]
Opening courses or directory makes a toast saying "No favorites", which shouldn't show up upon opening the fragment

**Comment by dlakata** [comment]
Clicking on a course in Courses opens up the detail view in the favorites tab

**Comment by dlakata** [comment]
make sure to keep spacing around if-else brackets consistent

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/SearchFavoriteFragment.java [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+
+        protected void setIndex(int indexkey, int arraykey, String query) {
+            SharedPreferences sharedPref = PreferenceManager.getDefaultSharedPreferences(getActivity());
+            int index = sharedPref.getInt(getString(indexkey), -1);
+            String[] previouskey = getResources().getStringArray(arraykey);
+            SharedPreferences.Editor editor = sharedPref.edit();
+            if (index != -1) {
+                boolean changed = false;
+                for (int i = 4; i >= 0; i--) {
+                    int id = (index + 5 - i) % 5;
```
Comment (dlakata) [inline PennMobile/src/main/java/com/pennapps/labs/pennmobile/SearchFavoriteFragment.java]: the number 5 seems to pop up a lot. make it a constant with a meaningful name

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/SearchFavoriteFragment.java [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+
+            }
+        });
+        mActivity.addTabs(getAdapter(), viewPager, false);
+        return v;
+    }
+
+    @Override
+    public void onListItemClick(ListView l, View v, int position, long id) {
+        String s = mListView.getAdapter().getItem(position).toString();
```
Comment (dlakata) [inline PennMobile/src/main/java/com/pennapps/labs/pennmobile/SearchFavoriteFragment.java]: redundant local variable

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/SearchFavoriteFragment.java [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+         * This function implements what the adapter will do given a query
+         * @param query the query passed in from the search view
+         * @return bol changed or not, the return value hasn't been used yet.
+         */
+        public abstract boolean onReceiveQuery (String query);
+
+        protected void setIndex(int indexkey, int arraykey, String query) {
+            SharedPreferences sharedPref = PreferenceManager.getDefaultSharedPreferences(getActivity());
+            int index = sharedPref.getInt(getString(indexkey), -1);
+            String[] previouskey = getResources().getStringArray(arraykey);
```
Comment (dlakata) [inline PennMobile/src/main/java/com/pennapps/labs/pennmobile/SearchFavoriteFragment.java]: camelCase

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/SearchFavoriteFragment.java [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+            public boolean onClose() {
+                searchSuggestion(false);
+                return false;
+            }
+        };
+        searchView.setOnQueryTextFocusChangeListener(focusListener);
+        searchView.setOnCloseListener(closeListener);
+    }
+
+    protected void searchSuggestion(boolean show) {
```
Comment (dlakata) [inline PennMobile/src/main/java/com/pennapps/labs/pennmobile/SearchFavoriteFragment.java]: perhaps consider breaking this into ```showSuggestions()``` and ```hideSuggestions()```

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/DirectoryTab.java [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+                    @Override
+                    public void call(final List<Person> people) {
+                        if (loadingPanel != null) {
+                            loadingPanel.setVisibility(View.GONE);
+                            if (people.isEmpty()) {
+                                if (no_results != null) {
+                                    no_results.setVisibility(View.VISIBLE);
+                                }
+                            } else {
+                                if (mListView != null) {
```
Comment (dlakata) [inline PennMobile/src/main/java/com/pennapps/labs/pennmobile/DirectoryTab.java]: move this up (```else if (mListView != null)```)

**Comment by jasontsh** [comment]
Two bugs that I'm trying to fix

- [ ] First time on courses, go to favorites, click on courses, go back to all, clicking on courses will not display fragment
- [ ] Back button should close the tab on the current page, not the last one of the page viewer

@dlakata @adelq @lilyidi opinions plox

**Pull request merged**